script/GameObjects/Elements/Components/Shield.lua

```diff
@@ -9,6 +9,9 @@
             take that energy from a capacitor. But physical objects, such as hull and
             armor, don't need to take energy and should have their own "health" values.
 --]]
+
+---@return Shield
```

Comment:
that one is a bit random

Reply:
So in making the effort to add Lua language server annotations as you asked me to do, it seemed correct to assert that `Shield` is returned when you create a new Shield component to add it to a vessel's components, because the Shield constructor returns an object of type... `Shield`?

For example, line 416 of StarSystem.lua receives a newly-created Shield component: `local shield = Components.Shield()` . And I tried to follow your example in StarSystem.lua itself, which is annotated with `---@return StarSystem` .

If I got something wrong with this, please let me know.

Reply:
yeah problem here is that this `---@return Shield` is in the middle of nowhere :D so it will not assign to anything. It´s fine i´ll fix it once i have the time to clean up the ECS a bit.